numpy/core/getlimits.py

```diff
@@ -208,23 +277,26 @@ def _register_known_types():
     # https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format#Double-double_arithmetic
     # These numbers have the same exponent range as float64, but extended number of
     # digits in the significand.
-    huge_dd = (umath.nextafter(ld(inf), ld(0))
-                if hasattr(umath, 'nextafter')  # Missing on some platforms?
-                else float64_ma.huge)
+    huge_dd = nextafter(ld(inf), ld(0), dtype=ld)
+    # As the smallest_normal in double double is so hard to calculate we set
+    # it to NaN.
+    smallest_normal_dd = NaN
     float_dd_ma = MachArLike(ld,
-                              machep=-105,
-                              negep=-106,
-                              minexp=-1022,
-                              maxexp=1024,
-                              it=105,
-                              iexp=11,
-                              ibeta=2,
-                              irnd=5,
-                              ngrd=0,
-                              eps=exp2(ld(-105)),
-                              epsneg= exp2(ld(-106)),
-                              huge=huge_dd,
-                              tiny=exp2(ld(-1022)))
+                             machep=-105,
+                             negep=-106,
+                             minexp=-1022,
+                             maxexp=1024,
+                             it=105,
+                             iexp=11,
+                             ibeta=2,
+                             irnd=5,
+                             ngrd=0,
+                             eps=exp2(ld(-105)),
+                             epsneg=exp2(ld(-106)),
+                             huge=huge_dd,
+                             tiny=smallest_normal_dd,
+                             smallest_normal=smallest_normal_dd,
+                             smallest_subnormal=exp2(ld(-16445)))
```

Comment:
@mattip do you remember what we said about the smallest subnormal here? 

If we use `np.nextafter` here, it might fail (platform dependend).  But we can actually use `ld(np.nextafter(0., 1.))` (the float64 version) since we know that is correct.   The value there looks like the extended precision float80 value, which is incorrect?

So I guess the problem is really just the `smallest_normal`, which technically was at `-1022`, but should be `-1022+54` if we want a full mantissa.

Reply:
I think we wanted to avoid long double, since it is platform dependent.